iree/compiler/Codegen/LLVMCPU/LLVMCPUTileFuseAndVectorizeLinalgTensorOps.cpp

```diff
@@ -54,11 +54,19 @@ struct TileWorkgroups : public linalg::LinalgBaseTilingPattern {
 namespace {
 struct LLVMCPUTileFuseAndVectorizePass
     : public LLVMCPUTileFuseAndVectorizeBase<LLVMCPUTileFuseAndVectorizePass> {
+  LLVMCPUTileFuseAndVectorizePass(bool vectorize = true)
```

Comment:
No need to do this in this CL, but you can just query the target and not vectorize if the target for the dispatch region is VMVX. This would make the decision local to the dispatch region and not global to the entire pass. Let me know if you are going to take care of this. I need something similar as well, so if you are adding this, I can rebase on that.

Reply:
I'm going to leave it this way (and could do cleanup in later PRs).

I think the scopes about "make the decision local to the dispatch region" and "global to the entire pass" are not aligned between us.

To query what we are target for the dispatch, we would go with querying a `hal.executable.variant`.

This is what I'm thinking, and I feel that this is not what you're expecting. The information can be queried in LLVMCPULowerExecutableTargetPass.

https://github.com/google/iree/blob/10e0193782c8260447862209ea087ef4702877d6/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp#L111

We are able to get target backend from the op and know if we want to vectorize it.

```mlir
hal.executable.variant public @vmvx_bytecode_fb, target = #hal.executable.target<"vmvx", "vmvx-bytecode-fb"> {
 ...
}
```

If we make the decision at this level, we still need this flag.

What we can do here is to remove the flag from `createLLVMCPULowerExecutableTargetPass`, not `LLVMCPUTileFuseAndVectorizePass`.

https://github.com/google/iree/blob/10e0193782c8260447862209ea087ef4702877d6/iree/compiler/Codegen/LLVMCPU/LLVMCPULowerExecutableTarget.cpp#L206-L208

WDYT?

Reply:
Querying the target from the `hal.executable.variant` is what I was thinking as well. Everything you are saying makes sense to me. I am indeed refering to removing the `lowerToVectors` option from `createLLVMCPULowerExecutableTargetPass` method.

Reply:
Got you. There are some cleanup needed in the tests. I'll send out another PR to address it.